homeassistant/components/sensor/entur_public_transport.py

```diff
@@ -0,0 +1,176 @@
+    ATTR_ATTRIBUTION, CONF_LATITUDE,
+    CONF_LONGITUDE, CONF_NAME, STATE_UNKNOWN)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+import homeassistant.util.dt as dt_util
+
+REQUIREMENTS = ['enturclient==0.1.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_EXPECTED_IN = 'due_in'
+ATTR_NEXT_UP_IN = 'next_due_in'
+
+API_CLIENT_ID = 'home-assistant'
+
+CONF_ATTRIBUTION = "Data provided by entur.org under NLOD."
+CONF_STOP_IDS = 'stop_ids'
+CONF_EXPAND_PLATFORMS = 'expand_platforms'
+CONF_ADD_TO_MAP = 'add_to_map'
+
+SCAN_INTERVAL = timedelta(minutes=1)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_STOP_IDS): vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_EXPAND_PLATFORMS, default=True): cv.boolean,
+    vol.Optional(CONF_ADD_TO_MAP, default=False): cv.boolean,
+})
+
+
+def due_in_minutes(timestamp: str) -> str:
+    """Get the time in minutes from a timestamp.
+
+    The timestamp should be in the format
+    year-month-yearThour:minute:second+timezone
+    """
+    if timestamp is None:
+        return STATE_UNKNOWN
+    diff = datetime.strptime(
+        timestamp, "%Y-%m-%dT%H:%M:%S%z") - dt_util.now()
+
+    return str(int(diff.total_seconds() / 60))
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Entur public transport sensor."""
+    from enturclient import EnturPublicTransportData
+    stop_ids = config.get(CONF_STOP_IDS)
+    expand = config.get(CONF_EXPAND_PLATFORMS)
+    add_to_map = config.get(CONF_ADD_TO_MAP)
+
+    stops = [s for s in stop_ids if "StopPlace" in s]
+    quays = [s for s in stop_ids if "Quay" in s]
+
+    data = EnturPublicTransportData(
+        API_CLIENT_ID,
+        stops,
+        quays,
+        expand)
+
+    proxy = EnturProxy(data)
+
+    data.update()
+    entities = []
+    for item in data.all_stop_places_quays():
+        entities.append(EnturPublicTransportSensor(proxy, item, add_to_map))
+
+    add_entities(entities, True)
+
+
+class EnturProxy:
+    """Proxy for the Entur client.
+
+    Ensure throttle to not hit rate limiting on api.
+    """
+
+    def __init__(self, data):
+        """Initialize the proxy."""
+        self._data = data
+
+    @Throttle(SCAN_INTERVAL)
+    def update(self) -> None:
+        """Update data in client."""
+        self._data.update()
+
+    def get_stop_info(self, stop_id: str) -> dict:
+        """Get info about specific stop place."""
+        return self._data.get_stop_info(stop_id)
+
+
+class EnturPublicTransportSensor(Entity):
+    """Implementation of a Entur public transport sensor."""
+
+    def __init__(self, data: EnturProxy, stop: str, add_to_map: bool):
+        """Initialize the sensor."""
+        self.data = data
+        self._stop = stop
+        self._add_to_map = add_to_map
+        self._times = data.get_stop_info(stop)
+        self._state = STATE_UNKNOWN
```

Comment:
Don't use `STATE_UNKNOWN`. Use `None` to represent unknown state.

Reply:
Removed the variable, as it wasn't used.